modular_skyrat/modules/customization/modules/client/preferences.dm

```diff
@@ -2163,13 +2170,47 @@ GLOBAL_LIST_INIT(food, list(
 					features["uses_skintones"] = !features["uses_skintones"]
 
 				if("erp_pref")
+					//SKYRAT EDIT ADDITION BEGIN - ERP_SLOT_SYSTEM
+					// User changed state of ERP pref
+					var/mob/living/carbon/human/M = user
+					var/mob/targetmob = usr
+					//SKYRAT EDIT ADDITION END
 					switch(erp_pref)
 						if("Yes")
 							erp_pref = "Ask"
+							//SKYRAT EDIT ADDITION BEGIN - ERP_SLOT_SYSTEM
+							// The user has set the ERP pref to a value other than "Yes", now we drop all items from ERP slots and can't use them
+							if(M.vagina != null)
+								M.dropItemToGround(M.vagina, TRUE, M.loc, TRUE, FALSE, TRUE)
+							if(M.anus != null)
+								M.dropItemToGround(M.anus, TRUE, M.loc, TRUE, FALSE, TRUE)
+							if(M.nipples != null)
+								M.dropItemToGround(M.nipples, TRUE, M.loc, TRUE, FALSE, TRUE)
+							if(M.penis != null)
+								M.dropItemToGround(M.penis, TRUE, M.loc, TRUE, FALSE, TRUE)
+							// If the user has an inventory of the ERP open, then we will hide it
+							if(usr.hud_used.ERP_inventory_shown && targetmob.hud_used)
+								usr.hud_used.ERP_inventory_shown = FALSE
+								usr.client.screen -= targetmob.hud_used.ERP_toggleable_inventory
+							// Find the ERP button of the inventory and make it invisible so that the user cannot interact with it
+							for(var/atom/movable/screen/human/ERP_toggle/E in targetmob.hud_used.static_inventory)
+								if(istype(E, /atom/movable/screen/human/ERP_toggle))
+									E.invisibility = 100
+							//SKYRAT EDIT ADDITION END
```

Comment:
Ditto.
```suggestion
							// The user has set the ERP pref to a value other than "Yes", now we drop all items from ERP slots and can't use them
							if(M.vagina != null)
								M.dropItemToGround(M.vagina, TRUE, M.loc, TRUE, FALSE, TRUE)
							if(M.anus != null)
								M.dropItemToGround(M.anus, TRUE, M.loc, TRUE, FALSE, TRUE)
							if(M.nipples != null)
								M.dropItemToGround(M.nipples, TRUE, M.loc, TRUE, FALSE, TRUE)
							if(M.penis != null)
								M.dropItemToGround(M.penis, TRUE, M.loc, TRUE, FALSE, TRUE)
							// If the user has an inventory of the ERP open, then we will hide it
							if(usr.hud_used.ERP_inventory_shown && targetmob.hud_used)
								usr.hud_used.ERP_inventory_shown = FALSE
								usr.client.screen -= targetmob.hud_used.ERP_toggleable_inventory
							// Find the ERP button of the inventory and make it invisible so that the user cannot interact with it
							for(var/atom/movable/screen/human/ERP_toggle/erp_button in targetmob.hud_used.static_inventory)
								if(istype(erp_button, /atom/movable/screen/human/ERP_toggle))
									erp_button.invisibility = 100
```
Also, I'm not sure if that `if(istype(erp_button, /atom/movable/screen/human/ERP_toggle))` is even needed, I recommend trying to remove it and only add it back if necessary.